openssl/src/crypto/pkcs5.rs

```diff
@@ -1,6 +1,42 @@
 use libc::c_int;
+use std::ptr::null;
+
+use crypto::symm_internal::evpc;
+use crypto::hash;
+use crypto::symm;
 use ffi;
 
+/// Derives a key and an IV from various parameters.
+/// New applications should use pbkdf2_hmac_sha1 or another more modern key
+/// derevation algorithm
+pub fn evp_bytestokey(typ: symm::Type, message_digest_type: hash::Type,
+                      salt: Option<&[u8]>, data: &[u8],
+                      count: u32) -> (Vec<u8>, Vec<u8>) {
```

Comment:
Could this return a struct with named fields rather than a pair? It seems like it'll be really easy to mix up the key and the IV otherwise.